src/App.jsx

```diff
@@ -1,86 +1,112 @@
+
 function App() {
-    const [todos, setTodos] = useState(()=> {
-        const storedTddos = localStorage.getItem('todos');
-        return storedTddos ? JSON.parse(storedTddos) : [{ text: 'Iniciar con buena actitud', completed: false }];
-    });
-    const [newTodo, setNewTodo] = useState('');
+  return <TodoList />;
+}
+
+function TodoList() {
+  const [todos, setTodos] = useState(() => {
+    try {
+      const storedTddos = localStorage.getItem("todos");
+      return storedTddos ? JSON.parse(storedTddos) : [initialTodo];
+    } catch (e) {
+      return [initialTodo];
+    }
+  });
+  const [newTodo, setNewTodo] = useState("");
 
+  useEffect(() => {
+    localStorage.setItem("todos", JSON.stringify(todos));
+  }, [todos]);
 
-    useEffect(() => {
-        localStorage.setItem('todos', JSON.stringify(todos));
-    }, [todos]);
+  const addTodo = () => {
+    if (newTodo.trim() === "") return;
+    setTodos([...todos, { text: newTodo, completed: false }]);
+    setNewTodo("");
+  };
 
-    const addTodo = () => {
-        if (newTodo.trim() === '') return;
-        setTodos([...todos, {text: newTodo, completed: false}]);
-        setNewTodo('');
-    };
+  const handleKeyDown = (e) => {
+    if (e.key === "Enter") {
+      addTodo();
+    }
+  };
 
-    const handleKeyDown = (e) => {
-        if (e.key === 'Enter') {
-            addTodo();
-        }
-    };
+  const toggleTodo = (index) => {
+    const updatedTodos = todos.map((todo, i) =>
+      i === index ? { ...todo, completed: !todo.completed } : todo
+    );
+    setTodos(updatedTodos);
+  };
 
-    const toggleTodo = (index) => {
-        const updatedTodos = todos.map((todo, i) =>
-            i === index ? {...todo, completed: !todo.completed} : todo
-        );
-        setTodos(updatedTodos);
-    };
+  const deleteTodo = (index) => {
+    const updatedTodos = todos.filter((_, i) => i !== index);
+    setTodos(updatedTodos);
+  };
 
-    const deleteTodo = (index) => {
-        const updatedTodos = todos.filter((_, i) => i !== index);
-        setTodos(updatedTodos);
-    };
+  return (
+    <div className="max-w-md mx-auto mt-10 p-5 bg-white shadow-lg rounded-lg">
+      <h1 className="text-2xl font-bold mb-4 text-center text-gray-700">
+        Lista de TODOs
+      </h1>
+      <div className="flex mb-4">
+        <input
+          value={newTodo}
+          onChange={(e) => setNewTodo(e.target.value)}
+          onKeyDown={handleKeyDown}
+          className="border p-2 rounded-l w-full"
+          placeholder="Añadir nueva tarea"
+        />
+        <button
+          onClick={addTodo}
+          className="bg-blue-500 text-white px-4 py-2 rounded-r"
+        >
+          Añadir
+        </button>
+      </div>
+      <ul>
+        {todos.map((todo, index) => (
+          <TodoItem
+            key={index}
+            todo={todo}
+            index={index}
+            toggleTodo={toggleTodo}
+            deleteTodo={deleteTodo}
```

Comment:
just a suggestion: right now there is this one big component, An easy aux component might be this `<TodoItem />` , just a suggestion